### PR TITLE
Format Python code with black and isort

### PR DESCRIPTION
**Pull request by cclauss**
Adds spellchecking and Python linting to the GitHub Action

`black --skip-string-normalization . `
* https://pypi.org/project/black

`isort --profile black .`
* https://pypi.org/project/isort

`codespell --count --skip="*.js,*.json"`
* https://pypi.org/project/codespell

`flake8 --builtins=Element,PyItemTemplate,PyListTemplate --max-complexity=10 --max-line-length=88`
* https://pypi.org/project/flake8

### CODE REVIEW

File: pyscriptjs/src/pyscript.py [hunk 1]
```diff
@@ -68,7 +72,7 @@ def format_mime(obj):
         format_dict, md_dict = mimebundle
     else:
         format_dict = mimebundle
-        md_dict = {}
+        md_dict = {}  # noqa: F841
```
Comment (barnwell) [inline pyscriptjs/src/pyscript.py:75]: `md_dict` can be removed if it's not used. We shouldn't add the `noqa` here.

**Review by barnwell — COMMENTED** [review]
Looks good to me. I left 2 minor comments.

File: pyscriptjs/examples/fractals.py [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+    width: int,
+    height: int,
+    *,
+    x: float = -0.5,
+    y: float = 0,
+    zoom: int = 1,
+    max_iterations: int = 100
+) -> np.array:
     """
-    From https://www.learnpythonwithrune.org/numpy-compute-mandelbrot-set-by-vectorization/.
```
[the remark this answers is not in the document]
Comment (cclauss) [reply pyscriptjs/examples/fractals.py:8]: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length

File: pyscriptjs/src/pyscript.py (hunk 1 above)
Comment (cclauss) [reply pyscriptjs/src/pyscript.py]: Removed.

File: pyscriptjs/examples/fractals.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cclauss) [reply pyscriptjs/examples/fractals.py:8]: We could put `From` on a separate line.

**Comment by jezdez** [comment]
@cclauss How would you feel about moving the linting config to pre-commit, so we make it easier to use locally, cross-platform and allow us to expand it to cover other types of code quality checks? /cc @mattkram

**Comment by pww217** [comment]
Added a PR for the pre-commit app to use `black`. Not familiar with isort but feel free to edit the config yaml.

https://github.com/pyscript/pyscript/pull/235

CC: @mattkram

**Comment by mattkram** [comment]
@cclauss Thank you for the PR and for prompting this discussion. As @pww217 mentioned, we're working to integrate these types of formatting checks and linting into `pre-commit`.

Thank you for introducing me to `codespell` too! It looks like there is a [`pre-commit` hook](https://pre-commit.com/hooks.html) available for that tool too, which I will investigate.

Tomorrow is FOSS Friday at Anaconda, and I will return to this PR after integrating some additional `pre-commit` hooks for any remaining items.

**Pull request closed without merging**